src/panels/lovelace/entity-rows/hui-media-player-entity-row.js

```diff
@@ -0,0 +1,129 @@
+import '@polymer/paper-icon-button/paper-icon-button.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+import '../components/hui-generic-entity-row.js';
+
+import LocalizeMixin from '../../../mixins/localize-mixin.js';
+
+const SUPPORT_PAUSE = 1;
+const SUPPORT_NEXT_TRACK = 32;
+const SUPPORTS_PLAY = 16384;
+const OFF_STATES = ['off', 'idle'];
+
+/*
+ * @appliesMixin LocalizeMixin
+ */
+class HuiMediaPlayerEntityRow extends LocalizeMixin(PolymerElement) {
+  static get template() {
+    return html`
+      <style>
+        .controls {
+          white-space: nowrap;
+        }
+      </style>
+      <hui-generic-entity-row
+        hass="[[hass]]"
+        config="[[_config]]"
+      >
+        <template is="dom-if" if="[[!_isOff(_stateObj.state)]]">
+          <div class="controls">
+            <template is="dom-if" if="[[_computeControlIcon(_stateObj)]]">
+              <paper-icon-button
+                icon="[[_computeControlIcon(_stateObj)]]"
+                on-click="_playPause"
+              ></paper-icon-button>
+            </template>
+            <template is="dom-if" if="[[_supportsNext(_stateObj)]]">
+              <paper-icon-button
+                icon="hass:skip-next"
+                on-click="_nextTrack"
+              ></paper-icon-button>
+            </template>
+          </div>
+        </template>
+        <template is="dom-if" if="[[_isOff(_stateObj.state)]]">
+          <div>[[_computeState(_stateObj.state)]]</div>
+        </template> 
+ 
+        <div slot="secondary">
+          [[_computeMediaTitle(_stateObj)]]
+        </div>
+      </hui-generic-entity-row>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: Object,
+      _config: Object,
+      _stateObj: {
+        type: Object,
+        computed: '_computeStateObj(hass.states, _config.entity)'
+      }
+    };
+  }
+
+  _computeStateObj(states, entityId) {
+    return states && entityId in states ? states[entityId] : null;
+  }
+
+  setConfig(config) {
+    if (!config || !config.entity) {
+      throw new Error('Entity not configured.');
+    }
+    this._config = config;
+  }
+
+  _computeControlIcon(stateObj) {
+    if (stateObj.state !== 'playing') {
+      return stateObj.attributes.supported_features & SUPPORTS_PLAY ? 'hass:play' : '';
+    }
+
+    return stateObj.attributes.supported_features & SUPPORT_PAUSE ? 'hass:pause' : 'hass:stop';
+  }
+
+  _computeMediaTitle(stateObj) {
+    if (!stateObj || this._isOff(stateObj.state)) return null;
+
+    switch (stateObj.attributes.media_content_type) {
```

Comment:
should we use hass-media-player-model.js? or maybe break up that model in helper functions

Reply:
I'm not sure what we should do here. That model class seems like it was tailored specifically for a larger card with multiple lines of information and lots of buttons, here we only have a couple buttons and a single line of text so we need a different media title format. 

The bigger question is do we even want to use the secondary_info for the media title? This is the same situtation as the climate entity-row since we are showing state information on the left side. I do kinda like it here, but if we decide that we are going to keep all state info on the right then maybe we should only show the entity name on the left and if people want more information about what's playing they can open more-info or use a media-control-card instead.

Reply:
I think it's fine for now. Let's just merge it.